x/bank/keeper/grpc_query.go

```diff
@@ -225,6 +225,28 @@ func (k BaseKeeper) DenomMetadata(c context.Context, req *types.QueryDenomMetada
 	}, nil
 }
 
+// DenomMetadataByQueryString is identical to DenomMetadata query, but receives request via query string.
+func (k BaseKeeper) DenomMetadataByQueryString(c context.Context, req *types.QueryDenomMetadataByQueryStringRequest) (*types.QueryDenomMetadataByQueryStringResponse, error) {
```

Comment:
Am I missine something but isn't this the same as `DenomMetadata`?

Reply:
If the issue is just the fact that we MUST provide a denom in the URL path, then I would avoid creating a duplicate query handler -- we can at least extract out the handler code into a single method.

In addition, is there no annotation or something that we can add to the existing query to allow an empty query argument?

Reply:
> we can at least extract out the handler code into a single method.

@alexanderbez  We need to suffice the BaseKeeper Interface, thus need to have this method with this specific request field and response field. What I can do is extract the duplicated method between `DenomMetadata` and DenomMetadataByQueryString` to reduce code duplication.

Reply:
Yes, the code within the query handlers are identical I believe, so that should be kept DRY 👍

Reply:
On the second look, the way it is right now seems as far as we could go for having non-duplicated code. 
We use the same abstracted keeper method, namely `k.GetDenomMetaData` whilst the remaining part is simply handling the request and response itself